src/main/java/ru/test/restservice/service/FileService.java

```diff
@@ -31,20 +32,13 @@ public static boolean isAllowedFile(String name) {
         return isTextFile(name) || isPicFile(name);
     }
 
-    @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
-    public static class FileException extends RuntimeException {
-        public FileException(String message) {
-            super(message);
-        }
-    }
-
     /**
      * Функция, возвращающая список файлов, с которыми пользователь может работать
      *
      * @param folder рабочая директория пользователя
      * @return Список объектов с информацией о файлах
      */
-    public static ArrayList<FileItemDTO> listFiles(String folder) throws IOException {
+    public ArrayList<FileItemDTO> listFiles(String folder) throws IOException {
         ArrayList<FileItemDTO> result = new ArrayList<>();
         List<Path> files = Files.list(Paths.get(folder)).filter(
                 (path) -> isAllowedFile(path.toString())).collect(Collectors.toList());
```

Comment:
Ты зачем-то коллектишь в лист и затем по нему итерируешься, хотя мог бы сразу начать фигачить, без цикла на следующей строчке
```suggestion
                (path) -> isAllowedFile(path.toString())).forEach ((file) -> {
```